src/transaction/types/input.rs

```diff
@@ -155,16 +181,24 @@ impl Input {
 
     pub const fn utxo_id(&self) -> &UtxoId {
         match self {
-            Self::Coin { utxo_id, .. } => utxo_id,
+            Self::CoinSigned { utxo_id, .. } => utxo_id,
+            Self::CoinPredicate { utxo_id, .. } => utxo_id,
             Self::Contract { utxo_id, .. } => utxo_id,
         }
     }
 
+    pub const fn contract_id(&self) -> Option<&ContractId> {
+        match self {
+            Self::Contract { contract_id, .. } => Some(contract_id),
+            _ => None,
+        }
+    }
+
     /// Return a tuple containing the predicate and its data if the input is of
     /// type `Coin`
```

Comment:
if the input is of type `CoinPredicate` ?

Reply:
https://github.com/FuelLabs/fuel-tx/pull/117/commits/b883896727de91e1013de8d6243f158ec30f7892

Reply:
Done! Lost the track of the git commit since it was rebased